src/test/java/uk/gov/hmcts/juror/api/moj/service/report/YieldPerformanceReportServiceImplTest.java

```diff
@@ -0,0 +1,196 @@
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.BALANCE;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.COMMENTS;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.CONFIRMED;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.COURT;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.DIFFERENCE;
+import static uk.gov.hmcts.juror.api.moj.controller.reports.response.YieldPerformanceReportResponse.TableHeading.REQUESTED;
+
+@SuppressWarnings({
+    "PMD.AssertionsShouldIncludeMessage",
+    "PMD.UnnecessaryFullyQualifiedName"
+})
+class YieldPerformanceReportServiceImplTest {
+
+    private final JurorPoolRepository jurorPoolRepository;
+    private final IPoolCommentRepository poolCommentRepository;
+    private final CourtQueriesRepository courtQueriesRepository;
+    private final YieldPerformanceReportService yieldPerformanceReportService;
+
+    public YieldPerformanceReportServiceImplTest() {
+        this.jurorPoolRepository = mock(JurorPoolRepository.class);
+        this.poolCommentRepository = mock(IPoolCommentRepository.class);
+        this.courtQueriesRepository = mock(CourtQueriesRepository.class);
+        this.yieldPerformanceReportService = new YieldPerformanceReportServiceImpl(jurorPoolRepository,
+            poolCommentRepository, courtQueriesRepository);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        TestUtils.setUpMockAuthentication("400", "BUREAU_USER", "1", List.of("400"));
+    }
+
+    @Nested
+    @DisplayName("Yield Performance report tests")
+    @SuppressWarnings("PMD.JUnitAssertionsShouldIncludeMessage") //false positive
+    class YieldPerformanceReportTests {
+
+        @Test
+        @SneakyThrows
+        void viewYieldPerformanceReportAllCourtsNoResultsAndValidHeadings() {
+
+            final LocalDate reportFromDate = LocalDate.of(2024, 4, 20);
+            final LocalDate reportToDate = LocalDate.of(2024, 5, 13);
+
+            YieldPerformanceReportRequest yieldPerformanceReportRequest = YieldPerformanceReportRequest.builder()
+                .fromDate(reportFromDate)
+                .toDate(reportToDate)
+                .allCourts(true)
+                .build();
+
+            when(courtQueriesRepository.getAllCourtLocCodes(false))
+                .thenReturn(List.of("415", "416"));
+
+            when(poolCommentRepository.findPoolCommentsForLocationsAndDates(List.of("415", "416"), reportFromDate,
+                reportToDate))
+                .thenReturn(List.of());
+
+            when(jurorPoolRepository.getYieldPerformanceReportStats("415,416", reportFromDate, reportToDate))
+                .thenReturn(List.of());
```

Comment:
Why do we return an empty list here?

Reply:
updated to null

Reply:
Why do we return null here?

Reply:
when there is nothing returned from database, it can be null, best to deal with that gracefully